common/store-api-sdk/src/mocks/local_kv_store.rs

```diff
@@ -0,0 +1,176 @@
+use common_metatypes::MatchSeq;
+use common_metatypes::MatchSeqExt;
+use common_metatypes::SeqValue;
+use common_store_api::kv_apis::kv_api::MGetKVActionResult;
+use common_store_api::GetKVActionResult;
+use common_store_api::PrefixListReply;
+use common_store_api::UpsertKVActionResult;
+
+use crate::KVApi;
+
+/// this mod is supposed to be used as a MOCK of LocalKVStoreApi
+/// crates depend on it, should explicitly enable the "mocks" feature of this crate
+/// like this:
+///
+/// [dev-dependencies]
+/// common-store-api-sdk= {path = "../store-api-sdk", features= ["mocks"]}
+///
+///
+type Key = String;
+type Value = SeqValue<KVValue>;
+
+#[derive(Default)]
+struct Inner {
+    // Store State machine uses one sequence for all keys, we simulated this
+    seq: u64,
+    kv: BTreeMap<Key, Value>,
+}
+
+impl Inner {
+    fn next_seq(&mut self) -> u64 {
+        self.seq += 1;
+        self.seq
+    }
+}
+
+/// A Simple in memory kv store
+pub struct LocalKVStore {
+    inner: RwLock<Inner>,
+}
+
+impl LocalKVStore {
+    #[allow(dead_code)]
+    pub fn new() -> common_exception::Result<LocalKVStore> {
+        let res = LocalKVStore {
+            inner: Default::default(),
+        };
+        Ok(res)
+    }
+}
+
+#[async_trait]
+impl KVApi for LocalKVStore {
+    async fn upsert_kv(
+        &self,
+        key: &str,
+        matcher: MatchSeq,
+        value: Option<Vec<u8>>,
+        value_meta: Option<KVMeta>,
+    ) -> Result<UpsertKVActionResult> {
+        let (prev, result) = {
+            let mut inner = self.inner.write();
+
+            if let Some((seq, v)) = inner.kv.get(key) {
+                let r = matcher.match_seq(*seq);
+                let prev = (*seq, v.clone());
+                if r.is_ok() {
+                    // seq match
+                    let mut new_v = v.clone();
+                    new_v.meta = value_meta;
+                    new_v.value = value.unwrap_or_default();
+                    let new_seq = inner.next_seq();
+                    let result = (new_seq, new_v);
+                    inner.kv.insert(key.to_owned(), result.clone());
+                    (Some(prev), Some(result))
+                } else {
+                    // seq not match,
+                    // store state machine use a pair of identical values to indicates
+                    // that nothing changed, we simulate it.
+                    let c = prev.clone();
+                    (Some(prev), Some(c))
+                }
+            } else {
+                // key not found, let's insert
+                let new_v = KVValue {
+                    meta: value_meta,
+                    value: value.unwrap_or_default(),
+                };
+                let new_seq = inner.next_seq();
+                let result = (new_seq, new_v);
+                inner.kv.insert(key.to_owned(), result.clone());
+                (None, Some(result))
+            }
+        };
+        Ok(UpsertKVActionResult { prev, result })
+    }
```

Comment:
A re-implementation of the meta-kv is not guaranteed to be consistent with the remote meta service.
Somebody who is not aware of there are two implementations may have changed one of them without updating the other one accordingly.
That's why a local kv should use a embedded `StateMachine`.

Reply:
> change log:
> 
> a. there are circular dependencies if we place StateMachine based LocalKVStore into crate store-api-sdk
> b. also, make common-xxx depend on crate store is not a good idea IMO
> c. let's rollback to the former implementations after the dependency issues resolved

Reply:
We both know about the reasons. A better way is to solve the dependency cycle first.

Reply:
I do not know what exactly the deps issues are before working on this pr.

Reply:
btw，maybe we should also try harder to NOT leak internal concepts like StateMachine to common-xxx directly

Reply:
> btw，maybe we should also try harder to NOT leak internal concepts like StateMachine to common-xxx directly

It seems we need a wrapper api of StateMachine?